homeassistant/components/reolink/host.py

```diff
@@ -359,6 +366,34 @@ def unregister_webhook(self):
         webhook.async_unregister(self._hass, self.webhook_id)
         self.webhook_id = None
 
+    async def _poll_all_motion(self, *_) -> None:
+        """Poll motion and AI states until the first ONVIF push is received."""
+        if self._webhook_reachable.is_set():
+            # ONVIF push is working, stop polling
+            self._cancel_poll = None
+            return
+
+        try:
+            await self._api.get_motion_state_all_ch()
+        except (
+            aiohttp.ClientConnectorError,
+            asyncio.TimeoutError,
+            ReolinkError,
+        ) as err:
+            _LOGGER.error(
+                "Reolink error while polling motion state for host %s:%s: %s",
+                self._api.host,
+                self._api.port,
+                str(err),
+            )
+
+        async_dispatcher_send(self._hass, f"{self.webhook_id}_all", {})
+
+        # schedule next poll
+        self._cancel_poll = async_call_later(
+            self._hass, POLL_INTERVAL_NO_PUSH, self._poll_all_motion
```

Comment:
You could wrap `self. _poll_all_motion` in `HassJob` so it doesn't have to be figured out how to call it every time.

You should also not schedule future polls if `hass.is_stopping` is `True`